modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,728 @@
+		}
+	}
+
+	return $tree;
+}
+
+/**
+ * Define an array of attribute value pairs for use inside the root element of an XML document.
+ * Intended for mapping namespace and namespace URI values.
+ * Passes array through jetpack_sitemap_ns for other functions to add their own namespaces.
+ *
+ * @module xml-sitemap
+ *
+ * @return array array of attribute value pairs passed through the jetpack_sitemap_ns filter
+ */
+function jetpack_sitemap_namespaces() {
+	/**
+	 * Filter the attribute value pairs used for namespace and namespace URI mappings.
+	 *
+	 * @module xml-sitemap
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $namespaces Associative array with namespaces and namespace URIs.
+	 */
+	return apply_filters( 'jetpack_sitemap_ns', array (
+		'xmlns:xsi'          => 'http://www.w3.org/2001/XMLSchema-instance',
+		'xsi:schemaLocation' => 'http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd',
+		'xmlns'              => 'http://www.sitemaps.org/schemas/sitemap/0.9',
+		// Mobile namespace from http://support.google.com/webmasters/bin/answer.py?hl=en&answer=34648
+		'xmlns:mobile'       => 'http://www.google.com/schemas/sitemap-mobile/1.0',
+		'xmlns:image'        => 'http://www.google.com/schemas/sitemap-image/1.1',
+	) );
+}
+
+/**
+ * Start sitemap XML document, writing its heading and <urlset> tag with namespaces.
+ *
+ * @module xml-sitemap
+ *
+ * @param $charset string Charset for current XML document.
+ *
+ * @return string
+ */
+function jetpack_sitemap_initstr( $charset ) {
+	// URL to XSLT
+	$xsl = get_option( 'permalink_structure' ) ? home_url( '/sitemap.xsl' ) : home_url( '/?jetpack-sitemap-xsl=true' );
+
+	$initstr = '<?xml version="1.0" encoding="' . $charset . '"?>' . "\n";
+	$initstr .= '<?xml-stylesheet type="text/xsl" href="' . esc_url( $xsl ) . '"?>' . "\n";
+	$initstr .= '<!-- generator="jetpack-' . JETPACK__VERSION . '" -->' . "\n";
+	$initstr .= '<urlset';
+	foreach ( jetpack_sitemap_namespaces() as $attribute => $value ) {
+		$initstr .= ' ' . esc_html( $attribute ) . '="' . esc_attr( $value ) . '"';
+	}
+	$initstr .= ' />';
+
+	return $initstr;
+}
+
+/**
+ * Load XSLT for sitemap.
+ *
+ * @module xml-sitemap
+ *
+ * @param string $type XSLT to load.
+ */
+function jetpack_load_xsl( $type = '' ) {
+
+	$transient_xsl = empty( $type ) ? 'jetpack_sitemap_xsl' : "jetpack_{$type}_sitemap_xsl";
+
+	$xsl = get_transient( $transient_xsl );
+
+	if ( $xsl ) {
+		header( 'Content-Type: ' . jetpack_sitemap_content_type(), true );
+		echo $xsl;
+		die();
+	}
+
+	// Populate $xsl. Use $type.
+	include_once JETPACK__PLUGIN_DIR . 'modules/xml-sitemap/sitemap-xsl.php';
+
+	if ( ! empty( $xsl ) ) {
+		set_transient( $transient_xsl, $xsl, DAY_IN_SECONDS );
+		echo $xsl;
+	}
+
+	die();
+}
+
+/**
+ * Responds with an XSLT to stylize sitemap.
+ *
+ * @module xml-sitemap
+ */
+function jetpack_print_sitemap_xsl() {
+	if ( defined( 'JETPACK_SKIP_DEFAULT_SITEMAP' ) && JETPACK_SKIP_DEFAULT_SITEMAP ) {
```

Comment:
What's the thought on using a constant over a filter here? I could see two outside authors trying to define it (a theme and an another plugin) and causing issues?

Applies to `JETPACK_SKIP_DEFAULT_NEWS_SITEMAP` below too.